packages/crypto/README.md

```diff
@@ -0,0 +1,94 @@
+| `sha256(<value>)`                   | Hashes a string/Uint8Array with SHA256                          | `Promise<Uint8Array>` | `WebCrypto`           |
+| `sha256Hex(<value>)`                | Same as `sha256` but reruns as hex string                       | `Promise<string>`     | `WebCrypto`           |
+| `sha256Sync(<value>)`               | Same as `sha256` but synchronous                                | `Uint8Array`          | `@noble/hashes`       |
+| `sha256SyncHex(<value>)`            | Same as `sha256Sync` but reruns as hex string                   | `string`              | `@noble/hashes`       |
+| `keccak256Sync(<value>)`            | Hashes a string/Uint8Array with Keccak256                       | `Uint8Array`          | `js-sha3`             |
+| `keccak256SyncHex(<value>)`         | Same as `keccak256Sync` but returns as hex string               | `string`              | `js-sha3`             |
+| `keccak256SyncHexPrefixed(<value>)` | Same as `keccak256Sync` but returns as `0x` prefixed hex string | `0x${string}`         | `js-sha3`             |
+
+## Initial Benchmarks
+
+#### sha256 -> Uint8Array
+
+```bash
+┌──────────────────────────────┬─────────┬────────────────────┬────────┬─────────┐
+│          Task Name           │ ops/sec │ Average Time (ns)  │ Margin │ Samples │
+├──────────────────────────────┼─────────┼────────────────────┼────────┼─────────┤
+│ @thirdweb-dev/crypto (async) │ 112,929 │ 8855.08539999982   │ ±1.37% │ 100000  │
+│ @thirdweb-dev/crypto (sync)  │ 55,004  │ 18180.482820000594 │ ±1.02% │ 100000  │
+│ crypto-js (sync)             │ 19,213  │ 52047.35353999857  │ ±0.25% │ 100000  │
+└──────────────────────────────┴─────────┴────────────────────┴────────┴─────────┘
+```
+
+#### sha256 -> hex
+
+```bash
+┌──────────────────────────────┬─────────┬────────────────────┬────────┬─────────┐
+│          Task Name           │ ops/sec │ Average Time (ns)  │ Margin │ Samples │
+├──────────────────────────────┼─────────┼────────────────────┼────────┼─────────┤
+│ @thirdweb-dev/crypto (async) │ 100,845 │ 9916.125730001713  │ ±1.24% │ 100000  │
+│ @thirdweb-dev/crypto (sync)  │ 53,751  │ 18604.047449997142 │ ±0.92% │ 100000  │
+│ crypto-js (sync)             │ 19,077  │ 52419.03465000334  │ ±0.24% │ 100000  │
+└──────────────────────────────┴─────────┴────────────────────┴────────┴─────────┘
+```
+
+#### keccack256 -> Uint8Array
+
+```bash
+┌──────────────────────┬─────────┬────────────────────┬────────┬─────────┐
+│      Task Name       │ ops/sec │ Average Time (ns)  │ Margin │ Samples │
+├──────────────────────┼─────────┼────────────────────┼────────┼─────────┤
+│ @thirdweb-dev/crypto │ 72,533  │ 13786.719900000808 │ ±0.31% │ 100000  │
+│ ethers@v5            │ 71,891  │ 13909.94663000107  │ ±0.33% │ 100000  │
+│ @noble/hashes        │ 27,244  │ 36704.73226000136  │ ±0.19% │ 100000  │
+│ viem                 │ 27,115  │ 36879.07953999698  │ ±0.26% │ 100000  │
```

Comment:
FYI, `@noble/hashes` can be faster, but it affects [auditability](https://github.com/paulmillr/noble-hashes/blob/main/benchmark/README.md).

Reply:
Again fair on the benchmarks. Similar to above.

While audibility is laudable and wanted, it coming at such a significant price of performance (even with such basic benchmarks) is currently not acceptable.

All of our users are currently using js-sha3 under the hood via ethers@5 so it is acceptable to go with the same solution for the moment (but make it easier for us to switch it out / have that control) because it's no longer wrapped 3 levels down. The end goal is not to stick with it for the long run. :)

Reply:
> significant price of performance

How many sha3 executions per second is "acceptable" for you? I don't think it matters for 99% of client apps AT ALL!

Reply:
I've published `unrolled-nbl-hashes-sha3` 1 year ago, which is the same stuff from noble but with shitty auditability and it runs around js-sha3 and others. What for? It's useless.

Reply:
Feel like the perf gain may _look_ large here, but hashing computation in the context of web3 libraries is generally constrained by other upstream dependant computation (ie. serializing, secp256k1, coding, network, etc). Hence, why I think they are misleading.